tools/shared.py

```diff
@@ -1240,7 +1241,8 @@ def is_exe(fpath):
     if fpath:
       if is_exe(program): return program
     else:
-      for path in os.environ["PATH"].split(os.pathsep):
+      paths = os.environ["PATH"].split(os.pathsep) + (extra_paths or [])
```

Comment:
I'd use
```python
if extra_paths is None:
  extra_paths = []
```
at the top of the function, to make the pattern clearer for future archaeologists.